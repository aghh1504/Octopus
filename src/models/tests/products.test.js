import ProductModel, { productFactory } from "../products";

describe("Products model", () => {
  it("should create an products with selected props", () => {
    const productsData = [
      {
        id: 1,
        image: {
          url: "",
          heigth: 510,
          width: 435
        },
        title: "Energy saving light bulb",
        details: "25W // Packet of 4",
        price: { amount: 12.99, currency: "Â£" },
        max_quantity: 10,
        description:
          "Available in 7 watts, 9 watts, 11 watts Spiral Light bulb in B22",
        specifications: {
          brand: "Philips",
          weight: "77g",
          dimensions: "12.6x6.2x6.2cm",
          model_number: "E27 ES",
          colour: "Cool daylight",
          type: "LED",
          socket: "B22/E27"
        },
        performance_first: [],
        performance_second: []
      }
    ];

    const products = productFactory(productsData);

    expect(products).toBeInstanceOf(ProductModel);

    expect(products).toMatchSnapshot();
  });
});
