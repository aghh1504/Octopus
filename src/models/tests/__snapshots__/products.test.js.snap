// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Products model should create an products with selected props 1`] = `
Immutable.ProductRecord {
  "id": null,
  "title": "",
  "image": Object {
    "heigth": 0,
    "url": "",
    "width": 0,
  },
  "details": "",
  "price": Object {
    "amount": 0,
    "currency": "",
  },
  "max_quantity": 0,
  "description": "",
  "specifications": Object {
    "brand": "",
    "colour": "",
    "dimensions": "",
    "model_number": "",
    "socket": "",
    "type": "",
    "weight": "",
  },
  "performance_first": Array [],
  "performance_second": Array [],
}
`;
